### DISCUSSION TITLE
Save user_id for Resource without BelongsToSelect?

### DISCUSSION THREAD

**Discussion by madsem** [filamentphp/filament#2149] [Help] [answered]
Hey,

is it possible to save a Resource where the Model has a `user` relationship, with the authenticated user_id without having to use a `BelongsToSelect`?

**Comment by madsem** [comment, marked as answer]
I found it myself, I think :)

```php
Forms\Components\Hidden::make('user_id')
                    ->dehydrateStateUsing(fn ($state) => auth()->user()->id),
```

**Reply by danharrin** [reply]
You could do this in the `mutateFormDataBeforeCreate()` method on the page :) its documented

**Reply by madsem** [reply]
Ahhh, apologies, I was looking all over the documentation haha. But I don't have the lingo down quite yet.
Method names are pretty technical. (And also Algolia didn't give me that example when searching for `user_id`)

That seems like a much cleaner way to go... And it indeed even has the exact example:
https://filamentphp.com/docs/2.x/admin/resources#customizing-data-before-saving

I was using more natural language to search for a solution. ;)

**Reply by danharrin** [reply]
Absolutely fine - hopefully this helps someone else

**Reply by khanakia** [reply]
@danharrin 
I tried using the above customizing data before saving the link and I used the exact same code but user_id still is saving as NULL . I have debugged the code and that created function executing fine but still, it's not saving. 

```
public function created(Post $post): void
{
    $post->user()->associate(auth()->user());
}
```

So how do i assign the current user_id automatically when the record is created?

**Reply by ryangjchandler** [reply]
Please make sure the column is fillable or your model is unguarded.

**Reply by danharrin** [reply]
Also, you need to call save() after associating
